src/components/Courses/Courses.jsx

```diff
@@ -1,68 +1,36 @@
-import React, { useState, useEffect } from 'react';
+// Courses.js
+import React, { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import SearchBar from './components/SearchBar/SearchBar';
 import CourseCard from './components/CourseCard/CourseCard';
 import Button from '../../common/Button/Button';
 import { API_ENDPOINTS, STORAGE_KEYS, PATHS } from '../../constants';
+import { fetchCourses } from '../../store/courses/actions'; // Adjust the import path as necessary
+import { fetchAuthors } from '../../store/authors/actions'; // Adjust the import path as necessary
 import './Courses.css';
 
 const Courses = ({ onAddCourseClick }) => {
-  const env = 'http://localhost:4000';
-  const [courses, setCourses] = useState([]);
-  const [authors, setAuthors] = useState([]);
   const [searchTerm, setSearchTerm] = useState('');
   const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const courses = useSelector((state) => state.courses);
+  const authors = useSelector((state) => state.authors);
 
   useEffect(() => {
     const token = localStorage.getItem(STORAGE_KEYS.USER_TOKEN);
```

Comment:
I assume the token should also be taken from Redux at this point (since Redux itself relies on Local Storage)